### PR TITLE
Remove nonblocking on flock, update comment.

### LINKED ISSUE #16096: E2e tests fail around "flock" command
Today I've seen multiple e2e failures due to a problem around ```flock``` command in https://github.com/kubernetes/kubernetes/blob/master/hack/jenkins/e2e.sh. AFAIU current code will fail (```-n``` flag) if another job is trying to update ```gcloud```. This means that an e2e test will be red until next scheduled run, which can be even 1h later. This blocks submit queue for a long time.

I think we should not use ```-n``` flag.

I'm not sending PR as the comment says only @brendandburns and @jlowdermilk can update this code :/

@wojtek-t @gmarek @davidopp

### PR DESCRIPTION
**Pull request by j3ffml**
Fixes #16096.

### CODE REVIEW

**Comment by jlowdermilk** [comment]
@k8s-bot test this

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -1121,27 +1121,33 @@ if [[ "${E2E_UP,,}" == "true" || "${JENKINS_FORCE_GET_TARS:-}" =~ ^[yY]$ ]]; the
...
           #
           # ssh into the jenkins master and run
-          # $ `sudo chown jenkins:jenkins /var/run/lock/gcloud-components.lock`
+          # $ sudo chown jenkins:jenkins /var/run/lock/gcloud-components.lock
+          #
+          # Note, flock -n would prevent parallel runs from having to wait
+          # here, but because we've set -o errexit, the err gets caught
+          # despite running in a subshell. If a run has to wait, the subsequent
+          # component update commands will be no-op, so no added delay.
+          flock -x 9
```
Comment (ixdy) [inline hack/jenkins/e2e.sh]: do we want to maybe set an explicit timeout? maybe 60s?

Comment (jlowdermilk) [inline hack/jenkins/e2e.sh]: Done.

**Comment by ixdy** [comment]
lgtm

**Comment by jlowdermilk** [comment]
@k8s-oncall consider merging manually to reduce e2e init failures.

**Comment by saad-ali** [comment]
Will merge on green.

**Comment by ixdy** [comment]
You can ignore Shippable for this PR - it isn't affecting anything Shippable would test.

**Pull request merged**